### ISSUE TITLE
Doesn't work for forked repos

### ISSUE THREAD

**Issue by SeanTAllen** [ponylang/changelog-bot-action#5]
on: pull_request events run in the forked repo's scope. not the base repo. Therefore, updates like what the changelog bot does can't be executed as they don't have access to push.

Probably what needs to happen to support forked repos...

- On pull request, collect the info we need and repost as an external webhook back to the base repo.
- Base repo runs the changelog update within its own scope.

This requires a more complicated setup, but, I think it will work.

**Comment by SeanTAllen** [comment]
I've figured out how to address this:

Listen for push events rather than pull request events. This will have us running in our own repo after the merge. However, we still need to get the PR info. We can do that by doing the following:

```
push event:

{
  repo: {
    name: "ponylang/ponyc"
  },
  payload: {
    head: "405a045a370af78611b8989467ff4123f81ee852"
  }
}
```

`head` is a commit that might be associated with a PR. to figure out if it is, let's do a search like:

```
curl https://api.github.com/search/issues?q=sha:4c079e2544b12d827beba12b1b2fc8ad45d662ab+is:merged+repo:ponylang/ponyc
```

That will return either:

```
{
  "total_count": 0
}
```

If empty. In which case, we are done. Because `total_count` is `0`. Otherwise, if it is 1, we will have:

```
{
  items [
    {
      pull_request: {
        url: https://api.github.com/repos/ponylang/ponyc/pulls/3372
      }
    }
  ]
}
```

where we can get the URL for the pull request and from that, all the info that we need including base ref (which branch to add CHANGELOG to, probably master) as well as the title, number, and changelog labels (if there are no labels, we skip).

**Comment by SeanTAllen** [comment]
Curl example to get PR from search:

`curl -s 'https://api.github.com/search/issues?q=is:merged+sha:1b47a73b985941d0f0252df05f86a7dd60bca083+repo:ponylang/ponyc' | jq -r '.items[].pull_request.url'`

returns:

`https://api.github.com/repos/ponylang/ponyc/pulls/3369` or nothing

**Comment by SeanTAllen** [comment]
Curl example to get needed information from PR event:

`curl -s "https://api.github.com/repos/ponylang/ponyc/pulls/3369" | jq -r '{pr: .number, change: .title, branch: .base.ref, labels: .labels[].name}'`
